core-bundle/src/Migration/Version503/RememberMeMigration.php

```diff
@@ -46,21 +46,24 @@ public function run(): MigrationResult
         );
 
         $schemaManager = $this->connection->createSchemaManager();
-        $username = isset($schemaManager->listTableColumns('tl_remember_me')['useridentifier']) ? 'userIdentifier' : 'username';
 
-        $this->connection->executeStatement(<<<SQL
-            INSERT INTO rememberme_token (
-                SELECT
-                    TRIM(TRAILING CHAR(0) FROM CAST(series AS char)),
-                    TRIM(TRAILING CHAR(0) FROM CAST(value AS char)),
-                    lastUsed,
-                    class,
-                    $username
-                FROM tl_remember_me
-                WHERE $username != ''
-            )
-            SQL,
-        );
+        // If tl_remember_me.userIdentifier exists, the database is from Contao 5 and the
+        // existing tokens can be migrated. Otherwise, it is a Contao 4 database and the
+        // tokens would not work anyway and therefore do not need to be migrated.
+        if (isset($schemaManager->listTableColumns('tl_remember_me')['useridentifier'])) {
+            $this->connection->executeStatement(<<<'SQL'
+                INSERT INTO rememberme_token (
+                    SELECT
+                        TRIM(TRAILING CHAR(0) FROM CAST(series AS char)),
+                        TRIM(TRAILING CHAR(0) FROM CAST(value AS char)),
+                        lastUsed,
+                        class,
+                        userIdentifier
+                    FROM tl_remember_me
```

Comment:
```suggestion
                    FROM tl_remember_me
                    WHERE userIdentifier != ''
```
We still need to check for empty `userIdentifier`s as these would possibly exist in a database that got updated in Contao 5.0-5.2